File: src/entity/bullet/RangeBullet.java
package entity.bullet;

import entity.enemy.Enemy;
import javafx.scene.image.Image;

public class RangeBullet extends Bullet {
	
	public static final int DAMAGE = 1;
	public static final int SPEED = 1;
	public static final String IMAGE_URL = "file:img/PNG/Retina/towerDefense_tile252.png";
	
	public RangeBullet(double x, double y, Enemy enemy) {
		super(x, y, enemy, SPEED, DAMAGE);
		
		img = new Image(IMAGE_URL);
	}
	
	@Override
	public void move() {
		double dx = enemy.getX() - getX();
		double dy = enemy.getY() - getY();
		
		double angle = Math.atan2(dy, dx);
		
		rotationAngle = Math.toDegrees(angle) + 90;
		
		setX(getX() + speed * Math.cos(angle));
		setY(getY() + speed * Math.sin(angle));
		
		speed += .05;
		
		if (speed > 9.8) {
			speed = 9.8;
		}
	}
	
}

File: src/entity/bullet/RangeBullet.kt
package entity.bullet

import entity.enemy.Enemy
import javafx.scene.image.Image

class RangeBullet(x: Double, y: Double, enemy: Enemy) : Bullet(x, y, enemy, RangeBullet.SPEED, RangeBullet.DAMAGE) {
    override var img: Image = Image(IMAGE_URL)

    init {
        flames = Image(FLAMES_IMAGE_URL)
    }

    override fun move() {

        super.move()

        speed += .05

        if (speed > 9.8) {
            speed = 9.8
        }
    }

    companion object {

        val DAMAGE = 1
        val SPEED = 1.0
        val IMAGE_URL = "file:img/PNG/Retina/towerDefense_tile252.png"
        val FLAMES_IMAGE_URL = "file:img/PNG/Retina/towerDefense_tile295.png"
    }

}
